### PR TITLE
about.yml generator (depends on shared-config)

### PR DESCRIPTION
**Pull request by yozlet**
Done with @msecret. Based on https://github.com/18F/about_yml . Note that since the `about.yml` file isn't used for much right now, we've skipped questions that seemed like too much work (mainly because they involved the user entering a list of items, e.g. URLs).

Also: this PR is branched from #49 but does not actually use any of the `shared-config` API. (We should probably discuss the best way to share config info.)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (cmc333333) [inline .eslintrc.json:1]: Not worth holding back this PR, but I think 18F's agreed to the airbnb lint guide.

File: about-yml/index.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+var Generator = require('yeoman-generator');
+
+var prompts = [
+  {
+    type: 'input',
+    name: 'repoName',
+    message: 'What\'s the repo name? (A short name that acts as the project identifier)'
+  },
```
Comment (cmc333333) [inline about-yml/index.js]: Optimization: We're using the current directory in the [npm generator](https://github.com/18F/18f-cli/blob/210eeff65e368ceb7e9ac9e253a3b6d5f1ed8513/npm/index.js#L23). Do you think that should be a shared prompt? Maybe one that defaults to the current dir?

File: about-yml/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+  {
+    type: 'input',
+    name: 'projectFullName',
+    message: 'What\'s the project title? (A few words, title cased, describing the project)'
+  },
```
Comment (cmc333333) [inline about-yml/index.js:20]: Optimization: We should probably pull this out and combine the text into a config shared with the [readme](https://github.com/18F/18f-cli/blob/c52e951aca74bf1d57404247a68cba68239f4a58/readme/index.js#L8) generator.

File: about-yml/index.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+module.exports = Generator.extend({
+  prompting: function() {
+    return this.prompt(prompts
+      .filter( (prompt) => {
+        if (this.config.get(prompt.name)) {
+          return false;
+        } else {
+          return true;
+        }
+      })
```
Comment (cmc333333) [inline about-yml/index.js]: Should this be simplified to
```js
.filter( prompt => !this.config.get(prompt.name))
```
or is there a nuance I'm missing?

File: about-yml/index.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      .map( (prompt) => {
+        if (prompt.type === 'input' && !prompt.default) {
+          prompt['default'] = '@TODO';
+        }
+        return prompt;
+      })
```
Comment (cmc333333) [inline about-yml/index.js]: This step feels out of place. I think we should either be explicit (though repetitive) within the prompts definition (maybe we use a constant for "@TODO"?) or this map step should be a `forEach` outside of the `prompting` phase. What do you think?

File: about-yml/index.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        }
+      });
+    });
+  },
+
+  writing: function () {
+    this.fs.copyTpl(
+      this.templatePath('about.yml'),
+      this.destinationPath('.about.yml'),
+      this.config.getAll()
```
Comment (cmc333333) [inline about-yml/index.js]: I like the concision!

**Review by cmc333333 — APPROVED** [review]
Added a few nits, but this seems like a very useful step forward.

File: about-yml/index.js (hunk 2 above)
Comment (msecret) [reply about-yml/index.js]: Yes definitely. I think in this case, this about-yml should make the prompt, defaulting to the directory and the npm generator should use the value set here (if it is set).

File: about-yml/index.js (hunk 4 above)
Comment (msecret) [reply about-yml/index.js]: Yes it should.

File: .eslintrc.json (hunk 1 above)
Comment (msecret) [reply .eslintrc.json:1]: Yea we should add it in, I think we just added something temporary for now.

File: about-yml/index.js (hunk 5 above)
Comment (msecret) [reply about-yml/index.js]: I could agree with that, as I just found that theres inputs (like repoName) where we don't want it to be a config, but want @yozlet to weigh in.

**Comment by cmc333333** [comment]
Pushed up a README and package.json change. Feel free to merge if that all looks good.

**Pull request merged**